### PR TITLE
Wait for OpenshiftVersion label when managed cluster is added

### PR DESCRIPTION
**Pull request by coleenquadros**
Added a condition to ensure that when vendor label is set to `openshift` and OpenShiftVersion is not we should not reconcile MCO to add observability stack in managed cluster
https://issues.redhat.com/browse/ACM-12249

### CODE REVIEW

File: operators/multiclusterobservability/controllers/placementrule/placementrule_controller.go [hunk 1]
```diff
@@ -591,7 +591,8 @@ func areManagedClusterLabelsReady(obj client.Object) bool {
 	if _, openshiftVersionPresent := obj.GetLabels()["openshiftVersion"]; openshiftVersionPresent && openshiftVendor {
 		return true
 	}
-	if vendorOk && vendor != "auto-detect" {
+
+	if vendorOk && vendor != "auto-detect" && vendor != "OpenShift" {
```
Comment (douglascamata) [inline operators/multiclusterobservability/controllers/placementrule/placementrule_controller.go:595]: Could we document what this function has to do in a comment/doc above it? It's written in a way that's very difficult to understand. I also suggest a small refactor: 

```go
func areManagedClusterLabelsReady(obj client.Object) bool {
	vendor, foundVendor := obj.GetLabels()["vendor"]
	if !foundVendor || vendor == "" {
		log.Info("ManagedCluster labels are not ready", "cluster", obj.GetName())
		return false
	}

	if vendor == "OpenShift" {
		_, foundOpenshiftVersion := obj.GetLabels()["openshiftVersion"]
		return foundOpenshiftVersion
	}

	if foundVendor && vendor != "auto-detect" {
		return true
	}


	log.Info("ManagedCluster labels are not ready", "cluster", obj.GetName())
	return false
}
```

Please double check that the logic I wrote in the function comment is correct.

Comment (douglascamata) [reply operators/multiclusterobservability/controllers/placementrule/placementrule_controller.go:595]: Hum, I ate a log line when there's no `openshiftVersion` label. Also could simplify even more:

```go
// areManagedClusterLabelsReady check if labels automatically set in the managed cluster
// are ready to be accessed. These labels are: "vendor" and "openshiftVersion".
// Labels are considered not ready when:
//
// - The "vendor" label isn't found.
// - The "vendor" label has value "auto-detect".
// - The "vendor" has value "openshift" but there is no "openshiftVersion".
//
func areManagedClusterLabelsReady(obj client.Object) bool {
	labels := obj.GetLabels()
	vendor, foundVendor := labels["vendor"]

	if !foundVendor || vendor == "" || vendor == "auto-detect" {
		log.Info("ManagedCluster labels are not ready", "cluster", obj.GetName())
		return false
	}

	if vendor == "OpenShift" {
		_, foundOpenshiftVersion := labels["openshiftVersion"]
		if !foundOpenshiftVersion {
			log.Info("ManagedCluster labels are not ready", "cluster", obj.GetName())
                        return false
		}
	}

	return true
}
```

**Review by thibaultmg — APPROVED** [review]
/lgtm

**Pull request merged**